File: Arrays/Java/PascalTriangle.java
//I tried to make a menu driven program to combine all the pascal triangle problems into a single file, 
//you can just copy-paste the particular method asked in the problem statement.

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

//Method to get a single element located at a aprticular row & column
public class PascalTriangle {
    public static int element(int row, int col) {
        int answer = 1;
        for(int i = 0; i<col; i++) {
            answer = answer * (row - i);
            answer /= (i + 1);
        }
        return answer;
    }

  //Method to generate an entire row of pascal triangle
    public static List<Integer> listElements(int row) {
        ArrayList<Integer> answer = new ArrayList<>();
        long temp = 1;
        answer.add(1);
        for(int col = 1; col < row; col++) {
            temp = temp * (row - col);
            temp /= col;
            answer.add((int) temp);
        }
        return answer;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("1 - Finding the element located at Row 'r' & Column 'c'\n" +
                "2 - Printing the entire Row of the triangle\n" +
                "3 - Printing the entire Pascal triangle");
        int choice = sc.nextInt();
        if(choice == 1) {
            System.out.println("Enter the row and column of the element :");
            int row = sc.nextInt();
            int col = sc.nextInt();
            System.out.println(element(row - 1, col - 1));
        }
        else if (choice == 2) {
            System.out.println("Enter the row number : ");
            int row = sc.nextInt();
            System.out.println(listElements(row));
        }
        else if(choice == 3) {
            List<List<Integer>> ans = new ArrayList<>();
            System.out.println("Enter the number of rows :");
            int row = sc.nextInt();
            for(int i = 1; i<=row; i++) {
                ans.add(listElements(i));
            }
            System.out.println(ans);
        }
    }
}

File: Arrays/CPP/PascalTriangle.cpp
//I tried to make a menu driven program to combine all the pascal triangle problems into a single file, 
//you can just copy-paste the particular method asked in the problem statement.

#include <iostream>
#include <vector>
using namespace std;

// Function to get a single element located at a particular row & column
int element(int row, int col) {
    int answer = 1;
    for (int i = 0; i < col; i++) {
        answer = answer * (row - i);
        answer /= (i + 1);
    }
    return answer;
}

// Function to generate an entire row of Pascal's Triangle
vector<int> listElements(int row) {
    vector<int> answer;
    long long temp = 1;
    answer.push_back(1); // Add the first element which is always 1
    for (int col = 1; col < row; col++) {
        temp = temp * (row - col);
        temp /= col;
        answer.push_back(static_cast<int>(temp));
    }
    return answer;
}

int main() {
    int choice;
    cout << "1 - Finding the element located at Row 'r' & Column 'c'\n"
         << "2 - Printing the entire Row of the triangle\n"
         << "3 - Printing the entire Pascal triangle\n";
    cin >> choice;

    if (choice == 1) {
        int row, col;
        cout << "Enter the row and column of the element: ";
        cin >> row >> col;
        cout << element(row - 1, col - 1) << endl;
    }
    else if (choice == 2) {
        int row;
        cout << "Enter the row number: ";
        cin >> row;
        vector<int> result = listElements(row);
        for (int num : result) {
            cout << num << " ";
        }
        cout << endl;
    }
    else if (choice == 3) {
        int numRows;
        cout << "Enter the number of rows: ";
        cin >> numRows;
        vector<vector<int>> ans;
        for (int i = 1; i <= numRows; i++) {
            ans.push_back(listElements(i));
        }
        for (const auto& row : ans) {
            for (int num : row) {
                cout << num << " ";
            }
            cout << endl;
        }
    }
    return 0;
}
